scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -19,13 +19,25 @@ function test_subctl_gather() {
 
   validate_pod_log_files $subm_ns $gateway_deployment_name
   validate_pod_log_files $subm_ns $routeagent_deployment_name
+
+  # Service Discovery
+  validate_resource_files all 'serviceexports.multicluster.x-k8s.io' 'ServiceExport'
+  validate_resource_files all 'serviceimports.multicluster.x-k8s.io' 'ServiceImport'
+  validate_resource_files all 'endpointslices.discovery.k8s.io' 'EndpointSlice'
+  validate_resource_files $subm_ns 'configmap submariner-lighthouse-coredns' 'ConfigMap'
```

Comment:
This doesn't work with a specific resource. First, the kubectl command:

`kubectl get configmap submariner-lighthouse-coredns --namespace=submariner-operator -o=jsonpath={.items..metadata.name}`

doesn't return the name. For getting a single resource, the jsonpath needs to be `{.metadata.name}`. Also `short_res=configmap submariner-lighthouse-coredns` which isn't correct. To handle a single resource, we need to change:

```
  names=$(kubectl get $resource --namespace=$ns -o=jsonpath='{.items..metadata.name}')
  read -ra names_array <<< "$names"

  short_res=$(echo $resource | awk -F. '{ print $1 }')
```

to:

```
  name_path=".items..metadata.name"
  res_name=$(echo $resource | awk '{ print $2 }')
  if [[ $res_name != "" ]]; then
    name_path=".metadata.name"
  fi

  names=$(kubectl get $resource --namespace=$ns $label_param -o=jsonpath="{$name_path}")
  read -ra names_array <<< "$names"

  short_res=$(echo $resource | awk '{ print $1 }')
  short_res=$(echo $resource | awk -F. '{ print $1 }')
```

Reply:
With the change you did for labels it is not needed, I can just use `-l`